app/models/manageiq/providers/vmware/cloud_manager/vm/operations.rb

```diff
@@ -1,3 +1,23 @@
 module ManageIQ::Providers::Vmware::CloudManager::Vm::Operations
+  extend ActiveSupport::Concern
+
   include_concern 'Power'
+
+  def raw_destroy
+    raise "VM has no #{ui_lookup(:table => "ext_management_systems")}, unable to destroy VM" unless ext_management_system
+    ext_management_system.with_provider_connection do |service|
+      begin
+        response = service.post_undeploy_vapp(ems_ref, :UndeployPowerAction => 'shutdown')
```

Comment:
Before delete action is called we need to stop VM. In case VM is already stopped action will rise an error. To successfully delete VM we are catching this error and just log the message what happened.